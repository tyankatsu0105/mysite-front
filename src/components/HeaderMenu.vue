<template>
	<div class="HeaderMenu">
		<nav class="HeaderMenu_Nav">
			<HeaderCloseButton @close-header="closeHeader" />
			<ul class="HeaderMenu_NavList">
				<li class="HeaderMenu_NavListItem">
					<g-link
						:to="{ name: 'home' }"
						class="HeaderMenu_NavListItemLink"
						exact
					>
						<span>
							Home
						</span>
					</g-link>
				</li>
				<li class="HeaderMenu_NavListItem">
					<g-link
						:to="{ name: 'about' }"
						class="HeaderMenu_NavListItemLink"
						exact
					>
						<span>
							About
						</span>
					</g-link>
					<g-link
						to="/blog/all"
						class="HeaderMenu_NavListItemLink"
						exact
					>
						<span>
							Blog
						</span>
					</g-link>
				</li>
			</ul>
		</nav>
	</div>
</template>

<script>
import HeaderCloseButton from "@/basics/HeaderCloseButton.vue";
export default {
  components: {
    HeaderCloseButton
  },
  props: {
    isOpenHeader: {
      type: Boolean
    }
  },
  methods: {
    closeHeader() {
      this.$emit("close-header", false);
    }
  }
};
</script>

<style scoped lang="scss">
.HeaderMenu {
  position: fixed;
  top: 0;
  left: 0;
  height: 100%;
  color: $color-primary;
  background-color: #fff;
  &_Nav {
    position: relative;
    height: 100vh;
    padding: 44px 0;
    overflow-y: scroll;
  }
  &_NavListItemLink {
    padding: 8px 40px;
    text-transform: uppercase;

    span {
      padding: 0 5px 5px;
      border-bottom: 1px solid transparent;
    }

    &.active {
      span {
        font-weight: bold;
        border-bottom-color: $color-accent;
      }
    }
  }
}
</style>
