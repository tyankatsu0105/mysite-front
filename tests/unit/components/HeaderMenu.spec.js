import { shallowMount } from "@vue/test-utils";
import HeaderMenu from "@/components/HeaderMenu.vue";

describe("HeaderMenu", () => {
  it("render", () => {
    const wrapper = shallowMount(HeaderMenu);

    expect(wrapper.html()).toMatchSnapshot();
  });
});
