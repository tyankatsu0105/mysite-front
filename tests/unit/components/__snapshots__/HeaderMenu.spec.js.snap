// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderMenu render 1`] = `
<div class="HeaderMenu">
  <nav class="HeaderMenu_Nav">
    <headerclosebutton-stub></headerclosebutton-stub>
    <ul class="HeaderMenu_NavList">
      <li class="HeaderMenu_NavListItem">
        <g-link-stub to="[object Object]" exact="" class="HeaderMenu_NavListItemLink"><span>
						Home
					</span></g-link-stub>
      </li>
      <li class="HeaderMenu_NavListItem">
        <g-link-stub to="[object Object]" exact="" class="HeaderMenu_NavListItemLink"><span>
						About
					</span></g-link-stub>
        <g-link-stub to="/blog/all" exact="" class="HeaderMenu_NavListItemLink"><span>
						Blog
					</span></g-link-stub>
      </li>
    </ul>
  </nav>
</div>
`;
